editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardRuntimeAndCamelPage.java

```diff
@@ -261,27 +275,28 @@ public void run() {
 	}
 	
 	private void configureRuntimeCombo() throws CoreException {
-		if (Widgets.isDisposed(runtimeCombo)) {
+		if (Widgets.isDisposed(runtimeComboViewer)) {
 			return;
 		}
-		int i =0, selectedRuntimeIdx = 0;
+		
 		String lastUsedRuntime = lastSelectedRuntime;
-
+		String selectedRuntime = null;
 		serverRuntimes = getServerRuntimes(null);
-		runtimeCombo.removeAll();
-		runtimeCombo.add(Messages.newProjectWizardRuntimePageNoRuntimeSelectedLabel);
-		runtimeCombo.select(0);
+		runtimeComboViewer.setInput(null);
+		runtimeComboViewer.add(Messages.newProjectWizardRuntimePageNoRuntimeSelectedLabel);
+		final ISelection selection = new StructuredSelection(Messages.newProjectWizardRuntimePageNoRuntimeSelectedLabel);
+		runtimeComboViewer.setSelection(selection);
 		for (Map.Entry<String, IRuntime> entry : serverRuntimes.entrySet()) {
-			runtimeCombo.add(entry.getKey());
-			++i;
+			runtimeComboViewer.add(entry.getKey());
```

Comment:
Optional: When using Viewers, we should not add directly entries to it.
There is setInput and then to calculate what is available, it is the role of the ContentProvider.
I won't block the PR for this point as it is still better than previously but if you can have a look and try to implement it following the state of the art it would be better.